catalog/task/create-internal-request/0.8/create-internal-request.yaml

```diff
@@ -0,0 +1,185 @@
+                if [ "${SOURCE}" == "sharedRequestFile" ]; then
+                    case ${TYPE} in
+                        "json")
+                            # it has a nested json string
+                            VALUE=`jq -cr "${KEY}" $(params.inputDataFile) \
+                            | jq -cr "${JSON_KEY}"`
+                        ;;
+                    esac
+                else
+                    VALUE=$(echo "${VALUE}" |tr -d "\\" 2>/dev/null |tr -d "\'" |jq -cr "${JSON_KEY}")
+                fi
+            fi
+            echo "    ${PARAM}: \"${VALUE}\"" >> ${IR}
+          done
+    - name: create-internal-request
+      image:
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      script: |
+          #!/usr/bin/env sh
+          PATH=/bin:/usr/bin:/usr/local/bin
+          export PATH
+
+          IR_DIR="$(workspaces.input.path)/$(params.subdirectory)/internal-request"
+          IR="${IR_DIR}/ir-$(params.pipelineRunName)-$(context.taskRun.uid).yaml"
+          kubectl create -f ${IR}
+    - name: watch-internal-request-status
+      image:
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      script: |
+          #!/usr/bin/env sh
+          PATH=/bin:/usr/bin:/usr/local/bin
+          TASKRUN="/tmp/$$.sh"
+
+          cat > ${TASKRUN} <<SH
+          #!/usr/bin/env sh
+          #
+          # the task might need to get input from a shared file
+          IR_DIR="$(workspaces.input.path)/$(params.subdirectory)/internal-request"
+          RESULTSFILE="\${IR_DIR}/$(params.pipelineRunName)-$(context.taskRun.uid)-results.txt"
+          echo -n \${RESULTSFILE} |tee $(results.requestResultsFile.path)
+
+          IR="ir-$(params.pipelineRunName)-$(params.request)"
+          while true; do
+            STATUS=\$(kubectl get internalrequest \${IR} -o jsonpath='{.status.conditions[0].reason}')
+            case "\${STATUS}" in
+              Succeeded | Failed )
+                echo "InternalRequest finished"
+                echo ${STATUS} | tee $(results.requestReason.path)
+                kubectl get internalrequest \${IR} -o jsonpath='{.status.conditions[0].message}' \
+                | tee $(results.requestMessage.path)
+                if [ "$(params.updateGenericResult)" == "true" ]; then
+                    kubectl get internalrequest \${IR} -o jsonpath='{.status.results.genericResult}' \
+                    | tee $(results.genericResult.path)
+                fi
+                kubectl get internalrequest \${IR} -o jsonpath='{.status.results}' \
+                | tee \${RESULTSFILE}
+                break
+                ;;
+              "*")
+                ;;
+            esac
+            sleep 30
+          done
+          [ \$(kubectl get internalrequest \${IR} -o jsonpath='{.status.conditions[0].status}') == "True" ]
```

Comment:
should be update too?

Reply:
This one does not need to change as it still changes to True only if it Suceeded. But in any case I can change to match with the other change.

Reply:
changed.